x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucketTests.java

```diff
@@ -80,12 +80,17 @@ public static Iterable<Object[]> parameters() {
         })));
     }
 
+    @SuppressWarnings("checkstyle:LineLength")
     private Expression build(Source source, Expression arg) {
         Literal from;
         Literal to;
         if (arg.dataType() == DataTypes.DATETIME) {
             from = new Literal(Source.EMPTY, new BytesRef("2023-02-01T00:00:00.00Z"), DataTypes.KEYWORD);
-            to = new Literal(Source.EMPTY, new BytesRef("2023-03-01T00:00:00.00Z"), DataTypes.KEYWORD);
+            to = new Literal(
+                Source.EMPTY,
+                DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis("2023-03-01T09:00:00.00Z"),
+                DataTypes.DATETIME
+            );
```

Comment:
I just realized you can mix and match these! Neat.

I think it'd probably be nice to do something like:
```
private Literal fromOrTo(String date) {
  if (randomBoolean() {
    return new Literal(Source.EMPTY, new BytesRef(date), randomBoolean() ? DataTypes.KEYWORD : DataTypes.TEXT);
  }
  return new Literal(
                Source.EMPTY,
                DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis(date),
                DataTypes.DATETIME
            )
}
```

and then call it for both the `from` and `to`. It'll make a random valid endpoint every time. If you stick the `@Repeat(iterations=100)` annotation on the class it'll run the whole thing 100 times. You can't commit that, but it's fine - just run it a few times locally to make sure all the random combinations work. Then remove the annotation and rerun the formatter.

From there on out the shear number of people and CI runs that run this will reproduce all the valid cases fairly frequently.

Reply:
That's a good idea!